cpp/cmake_modules/FindArrow.cmake

```diff
@@ -1,466 +0,0 @@
-    arrow_extract_macro_value(ARROW_VERSION_MAJOR "ARROW_VERSION_MAJOR"
-                              "${ARROW_CONFIG_H_CONTENT}")
-    arrow_extract_macro_value(ARROW_VERSION_MINOR "ARROW_VERSION_MINOR"
-                              "${ARROW_CONFIG_H_CONTENT}")
-    arrow_extract_macro_value(ARROW_VERSION_PATCH "ARROW_VERSION_PATCH"
-                              "${ARROW_CONFIG_H_CONTENT}")
-    if("${ARROW_VERSION_MAJOR}" STREQUAL ""
-       OR "${ARROW_VERSION_MINOR}" STREQUAL ""
-       OR "${ARROW_VERSION_PATCH}" STREQUAL "")
-      set(ARROW_VERSION "0.0.0")
-    else()
-      set(ARROW_VERSION
-          "${ARROW_VERSION_MAJOR}.${ARROW_VERSION_MINOR}.${ARROW_VERSION_PATCH}")
-    endif()
-
-    arrow_extract_macro_value(ARROW_SO_VERSION_QUOTED "ARROW_SO_VERSION"
-                              "${ARROW_CONFIG_H_CONTENT}")
-    string(REGEX REPLACE "^\"(.+)\"$" "\\1" ARROW_SO_VERSION "${ARROW_SO_VERSION_QUOTED}")
-    arrow_extract_macro_value(ARROW_FULL_SO_VERSION_QUOTED "ARROW_FULL_SO_VERSION"
-                              "${ARROW_CONFIG_H_CONTENT}")
-    string(REGEX REPLACE "^\"(.+)\"$" "\\1" ARROW_FULL_SO_VERSION
-                         "${ARROW_FULL_SO_VERSION_QUOTED}")
-  endif()
-else()
-  if(ARROW_USE_CMAKE_PACKAGE_CONFIG)
-    find_package(Arrow CONFIG)
-  elseif(ARROW_USE_PKG_CONFIG)
-    pkg_get_variable(ARROW_SO_VERSION arrow so_version)
-    pkg_get_variable(ARROW_FULL_SO_VERSION arrow full_so_version)
-  endif()
-endif()
-
-set(ARROW_ABI_VERSION ${ARROW_SO_VERSION})
-
-mark_as_advanced(ARROW_ABI_VERSION
-                 ARROW_CONFIG_SUFFIXES
-                 ARROW_FULL_SO_VERSION
-                 ARROW_IMPORT_LIB
-                 ARROW_INCLUDE_DIR
-                 ARROW_LIBS
-                 ARROW_LIB_DIR
-                 ARROW_SEARCH_LIB_PATH_SUFFIXES
-                 ARROW_SHARED_IMP_LIB
-                 ARROW_SHARED_LIB
-                 ARROW_SO_VERSION
-                 ARROW_STATIC_LIB
-                 ARROW_VERSION
-                 ARROW_VERSION_MAJOR
-                 ARROW_VERSION_MINOR
-                 ARROW_VERSION_PATCH)
-
-find_package_handle_standard_args(
-  Arrow
-  REQUIRED_VARS # The first required variable is shown
-                # in the found message. So this list is
-                # not sorted alphabetically.
-                ARROW_INCLUDE_DIR ARROW_LIB_DIR ARROW_FULL_SO_VERSION ARROW_SO_VERSION
-  VERSION_VAR ARROW_VERSION)
-set(ARROW_FOUND ${Arrow_FOUND})
-
-if(Arrow_FOUND AND NOT Arrow_FIND_QUIETLY)
-  message(STATUS "Arrow version: ${ARROW_VERSION} (${ARROW_FIND_APPROACH})")
-  message(STATUS "Arrow SO and ABI version: ${ARROW_SO_VERSION}")
-  message(STATUS "Arrow full SO version: ${ARROW_FULL_SO_VERSION}")
-  message(STATUS "Found the Arrow core shared library: ${ARROW_SHARED_LIB}")
-  message(STATUS "Found the Arrow core import library: ${ARROW_IMPORT_LIB}")
-  message(STATUS "Found the Arrow core static library: ${ARROW_STATIC_LIB}")
```

Comment:
If I understand correctly, this file is now replaced by cmake's `find_package(Arrow)` now using the `ArrowConfig.cmake` file that gets installed in the cmake directory (when installing Arrow C++). 
But so before, the above printed some details about Arrow being found (and which version, the path, ..). Is there a way to let `find_package(Arrow)` still print this? (would adding that to `ArrowConfig.cmake.in` do this?)

Reply:
Yes. We need to add `message(STATUS ...)` to `ArrowConfig.cmake.in` for it. `cmake --debug-print` prints more details but it will not show these information. 

But `*Config.cmake` doesn't use `message(STATUS ...)` in general. If we want to add `message(STATUS ...)`, it's better that we check `Arrow_FIND_QUIETLY` like the following:

```cmake
if(NOT Arrow_FIND_QUIETLY)
  message(STATUS ...)
endif()
```

`Arrow_FIND_QUIETLY` is `TRUE` only when an user specify `QUIET` explicitly like `find_package(Arrow QUIET)`.

See also; https://cmake.org/cmake/help/latest/command/find_package.html#package-file-interface-variables

Reply:
I've created https://issues.apache.org/jira/browse/ARROW-17632 to follow this up, feel free to update the ticket if I missed anything

Reply:
Thanks @kou for the explanation and @raulcd for opening the JIRA!